modin/experimental/engines/omnisci_on_ray/io.py

```diff
@@ -288,3 +343,130 @@ def _prepare_pyarrow_usecols(cls, read_csv_kwargs):
+        implementation.
+
+        Parameters
+        ----------
+        read_csv_kwargs: ReadCsvKwargsType
+                Parameters of read_csv function.
+
+        Returns
+        -------
+        bool:
+            Whether passed parameters are supported or not.
+        str:
+            Error message that should be raised if user explicitly set `engine="arrow"`.
+        """
+        filepath_or_buffer = read_csv_kwargs.get("filepath_or_buffer", None)
+        header = read_csv_kwargs.get("header", "infer")
+        names = read_csv_kwargs.get("names", None)
+        engine = read_csv_kwargs.get("engine", None)
+        skiprows = read_csv_kwargs.get("skiprows", None)
+        delimiter = read_csv_kwargs.get("delimiter", None)
+
+        if read_csv_kwargs.get("compression", "infer") != "infer":
+            return (
+                False,
+                "read_csv with 'arrow' engine doesn't support compression parameter, compression inferred"
+                " automatically (supported compression types are gzip and bz2)",
+            )
+
+        if isinstance(filepath_or_buffer, str):
+            if not os.path.exists(filepath_or_buffer):
+                from pandas.io.common import is_url
+
+                if cls.file_exists(filepath_or_buffer) or is_url(filepath_or_buffer):
+                    return (
+                        False,
+                        "read_csv with 'arrow' engine supports only local files",
+                    )
+                else:
+                    raise FileNotFoundError("No such file or directory")
+        elif not cls.pathlib_or_pypath(filepath_or_buffer):
+            if hasattr(filepath_or_buffer, "read"):
+                return (
+                    False,
+                    "read_csv with 'arrow' engine doesn't support file handles",
+                )
+            else:
+                raise ValueError(
+                    f"Invalid file path or buffer object type: {type(filepath_or_buffer)}"
+                )
+
+        read_csv_signature = inspect.signature(cls.read_csv)
+        read_csv_default_kwargs = {
+            k: v.default
+            for k, v in read_csv_signature.parameters.items()
+            if v.default is not inspect.Parameter.empty
+        }
+        for arg in cls.unsupported_args:
+            if read_csv_kwargs[arg] != read_csv_default_kwargs[arg]:
+                return (
+                    False,
+                    f"read_csv with 'arrow' engine doesn't support {arg} parameter",
+                )
+        if delimiter is not None and read_csv_kwargs.get("delim_whitespace", False):
+            raise ValueError(
+                "Specified a delimiter with both sep and delim_whitespace=True; you can only specify one."
+            )
+        elif delimiter == "\n":
+            return (
+                False,
+                "read_csv with 'arrow' engine doesn't support delimiter = '\n' parameter",
```

Comment:
```suggestion
                "read_csv with 'arrow' engine doesn't support delimiter = '\\n' parameter",
```
if this message will be printed out somewhere `\n` will be interpreted as an actual new line character